lib/ruby_lsp/document.rb

```diff
@@ -159,6 +160,10 @@ def locate(node, char_position, node_types: [])
           nesting << candidate
         end
 
+        if candidate.is_a?(Prism::CallNode)
+          call_node = candidate
```

Comment:
I feel there may be situations that aren't properly handled with this approach, but haven't found a concrete one yet. 🤔

Reply:
Added a test and fixed the behaviour.